### PR TITLE
games-arcade/*, games-board/*: EAPI bumps, various fixes

### PR DESCRIPTION
**Pull request by mm1ke**
Hi,

This updates a few games to `EAPI 7`.
I've also fixed the open bugs for `games-board/freedoko`. Turns out it had simply the wrong dependencies. Also fixed a few `gcc` & `g++` directly calls.

Please review.
@thesamesam @ionenwks you might be interested here :)

### CODE REVIEW

**Comment by mm1ke** [comment]
> ## Pull request CI report
> 
> _Report generated at_: 2021-06-09 18:29 UTC
> _Newest commit scanned_: [8d185ef](https://github.com/gentoo/gentoo/commit/8d185ef1eb69c9e27cf4266f3920f5e37fcad7d2)
> _Status_: x **broken**
> 
> New issues caused by PR:
> https://qa-reports.gentoo.org/output/gentoo-ci/22762946f9/output.html#games-board/pasang-emas
> 
> There are existing issues already. Please look into the report to make sure none of them affect the packages in question:
> https://qa-reports.gentoo.org/output/gentoo-ci/22762946f9/output.html

Can someone tell me what's wrong here? Unfortunately i cannot open the CI link :/

**Comment by ionenwks** [comment]
> Can someone tell me what's wrong here? Unfortunately i cannot open the CI link :/

`pasang-emas-3.1.0-r1 MissingSlash	${D} missing trailing slash on line: 57`

While the global check can pick up additional things, these could easily be checked locally by using pkgcheck yourself, e.g. `pkgcheck scan --commits -c=+InheritsCheck` is pretty handy.

**Comment by mm1ke** [comment]
Thanks @ionenwks for the info.
I've fixed the issue now. Also the CI-Link works now too. I knew it could take a bit until it works but didn't expected it to take so long :D

**Review by ionenwks — COMMENTED** [review]
Just an initial look, this is starting to get into the uglier end of games@ and some of those packages could use at least a bit of extra attention if we're going to touch them.

File: games-arcade/kajaani-kombat/kajaani-kombat-0.7-r1.ebuild [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
 
 EAPI=7
-inherit desktop toolchain-funcs
 
-DESCRIPTION="A rampart-like game set in space"
+inherit toolchain-funcs
+
+DESCRIPTION="Rampart-like game set in space"
 HOMEPAGE="http://kombat.kajaani.net/"
 SRC_URI="http://kombat.kajaani.net/dl/${P}.tar.gz"
```
[the remark this answers is not in the document]
Comment (mm1ke) [reply games-arcade/kajaani-kombat/kajaani-kombat-0.7-r1.ebuild]: I saw the bug actually but haven't looked into. I could give it a try but i guess i wouldn't be able to find a suitable fix for this.

**Comment by mm1ke** [comment]
@ionenwks 
Thanks for the detailed review. I hope i mostly fixed the issues you mentioned (and sorry for the delay.. my weekend was quite exhausting ;) ). 
FYI: I also additionally moved the `sys-devel/gettext` dependency from `games-board/pasang-emas` into `BDEPEND`.

File: games-arcade/kajaani-kombat/kajaani-kombat-0.7-r1.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (ionenwks) [reply games-arcade/kajaani-kombat/kajaani-kombat-0.7-r1.ebuild]: No biggie, I fixed it locally now.

I'll add a commit for it after the rest when I merge this a bit later.

**Comment by ionenwks** [comment]
Thanks.

Pushed mostly as-is but amended:
- freedoko: previous cppflags used `='"content"'` meaning they included `"` to be inlined by cpp, changing it to `="content"` leads to build failure, so I re-added them.
- pasang-emas: moved app-text/rarian to BDEPEND that I missed earlier

I did some additional commits for more cleanup/fixes while here, including the aforementioned kajaani's gcc11 fix.

**Pull request closed without merging**